mobile/library/cc/engine_builder.cc

```diff
@@ -259,26 +259,46 @@ EngineBuilder& EngineBuilder::enforceTrustChainVerification(bool trust_chain_ver
   return *this;
 }
 
-EngineBuilder& EngineBuilder::addRtdsLayer(const std::string& layer_name,
-                                           const int timeout_seconds) {
-  rtds_layer_name_ = layer_name;
-  rtds_timeout_seconds_ = timeout_seconds;
+EngineBuilder& EngineBuilder::setNodeId(std::string node_id) {
+  node_id_ = std::move(node_id);
+  return *this;
+}
+
+EngineBuilder& EngineBuilder::setNodeLocality(std::string region, std::string zone,
+                                              std::string sub_zone) {
+  set_node_locality_ = true;
+  node_locality_region_ = std::move(region);
+  node_locality_zone_ = std::move(zone);
+  node_locality_sub_zone_ = std::move(sub_zone);
   return *this;
 }
-EngineBuilder& EngineBuilder::setAggregatedDiscoveryService(const std::string& api_type,
-                                                            const std::string& address,
-                                                            const int port) {
+
+EngineBuilder& EngineBuilder::setAggregatedDiscoveryService(const std::string& address,
+                                                            const int port, std::string jwt_token,
+                                                            const int jwt_token_lifetime_seconds,
+                                                            std::string ssl_root_certs) {
 #ifndef ENVOY_GOOGLE_GRPC
   throw std::runtime_error("google_grpc must be enabled in bazel to use ADS");
 #endif
-  ads_api_type_ = api_type;
   ads_address_ = address;
   ads_port_ = port;
+  jwt_token_ = std::move(jwt_token);
+  jwt_token_lifetime_seconds_ = jwt_token_lifetime_seconds;
+  ssl_root_certs_ = std::move(ssl_root_certs);
```

Comment:
My thinking was to prefix all the ADS config variables with `ads_`, and `rtds_` for the RTDS ones.  Just to avoid naming conflicts as this file grows.

Reply:
done